pytorch_lightning/trainer/training_loop.py

```diff
@@ -99,7 +100,8 @@ def should_skip_training(self) -> bool:
         return should_by_max_steps or should_by_epoch or self.trainer.num_training_batches == 0
 
     def on_train_start(self):
-        # hook
+        self.trainer.results.to(device=self.trainer.lightning_module.device)
```

Comment:
Does this change the device internally ?

Reply:
yes:

https://github.com/PyTorchLightning/pytorch-lightning/blob/764d2c775e2f3568975a60e94520bca8a42b2490/pytorch_lightning/trainer/connectors/logger_connector/result_new.py#L495-L496